### PR TITLE
--workdir changed to -D to increase compatibility

### PR DESCRIPTION
**Pull request by pauloestrela**
After 17.02 slurm's sbatch command changed the --workdir switch to --chdir, but the short version (-D) remained the same. So, to maintain compatibility with older and newer versions, a better approach is to use -D instead of --workdir.

**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by ertomatic** [comment]
Can one of the admins verify this patch?

**Comment by oyvindeide** [comment]
Hi and thanks for the patch! I see the style check is failing, there is a convenience script at `/scripts/clang-format`.

File: src/clib/lib/job_queue/slurm_driver.cpp [hunk 1]
```diff
@@ -409,7 +409,7 @@ void *slurm_driver_submit_job(void *__driver, const char *cmd, int num_cpu,
     auto submit_script =
         make_submit_script(driver, cmd, job_name, num_cpu, argc, argv);
     std::vector<std::string> sbatch_argv = {
-        "--workdir=" + std::string(run_path),
+        "-D=" + std::string(run_path),
```
Comment (verveerpj) [inline src/clib/lib/job_queue/slurm_driver.cpp:412]: This does not work on the TNO SLUM clusters.

I tested `--chdir` on two SLURM systems that we use at TNO, and that works fine.

Hence, I suggest changing it to `--chdir`.

**Review by verveerpj — CHANGES_REQUESTED** [review]

**Comment by pauloestrela** [comment]
Which slurm version does TNO slurm use? Because --chdir do not work on versions below 17.02.

Obter o Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by verveerpj** [comment]
We use slurm 18.08.8

**Comment by pauloestrela** [comment]
@verveerpj Could you please try again? I accidentally left a = sign after -D instead of a space character.

**Comment by oysteoh** [comment]
> @verveerpj Could you please try again? I accidentally left a = sign after -D instead of a space character. I've already committed this change in my fork.

@pauloestrela Could you please also fix the formatting by running the script mentioned by @oyvindeide ?

**Comment by pauloestrela** [comment]
Sure!

**Comment by verveerpj** [comment]
Using a space instead if `=` does not work, however you can do this: `"-D", + std::string(run_path)`, i.e. no space and split the string with a comma. I have tested that and it works fine.

**Comment by oysteoh** [comment]
And please make sure when making the last change to also squash your commits together into one to leave a clean history when merged 👍

**Comment by oyvindeide** [comment]
@pauloestrela do you have time to update according to the findings in the discussion? Just let us know if we should assist (will keep you as author of course). Think this is a really good contribution, and would be good to include it.

**Comment by pauloestrela** [comment]
@oyvindeide Yes, I'll to that. I'm little busy right know working with Ketil to make sure that FMU and Drogon model are running well in our environment. Today at afternoon or maybe tomorrow morning I'll work on this PR.

**Comment by oyvindeide** [comment]
No worries, do it when you have time, and don't hesitate to reach out if you have any questions or need help with anything.

**Comment by oyvindeide** [comment]
Hi again @pauloestrela, did you have time to look at this?

**Pull request closed without merging**